### ISSUE TITLE
Trying to compile openssh.

### ISSUE THREAD

**Issue by txdv** [Alexpux/MSYS2-packages#31]
So I tried to compile openssh, i went into the directory and just typed: makedpkg

but I get this error when trying to compile: http://pastebin.com/rZHZbSe6

it stops during the configuration process:

```
checking if gcc supports compile flag -Wno-attributes... yes
checking compiler and flags for sanity... no
configure: error: *** compiler cannot create working executables, check config.log ***
==> ERROR: A failure occurred in build().
    Aborting...
```

I am using a 64bit windows install, latest msys2 for 64 bit and ming64-gcc:

```
$ pacman -Ss gcc
mingw32/mingw-w64-i686-gcc 4.9.1-3 (mingw-w64-i686-toolchain mingw-w64-i686) [installed]
    GNU Compiler Collection (C,C++,OpenMP) for MinGW-w64
mingw32/mingw-w64-i686-gcc-ada 4.9.1-3 (mingw-w64-i686-toolchain mingw-w64-i686)
    GNU Compiler Collection (Ada) for MinGW-w64
mingw32/mingw-w64-i686-gcc-ada-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-i686-gcc-ada
mingw32/mingw-w64-i686-gcc-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-i686-gcc
mingw32/mingw-w64-i686-gcc-fortran 4.9.1-3 (mingw-w64-i686-toolchain mingw-w64-i686)
    GNU Compiler Collection (Fortran) for MinGW-w64
mingw32/mingw-w64-i686-gcc-fortran-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-i686-gcc-fortran
mingw32/mingw-w64-i686-gcc-libgfortran 4.9.1-3 (mingw-w64-i686-toolchain mingw-w64-i686)
    GNU Compiler Collection (libgfortran) for MinGW-w64
mingw32/mingw-w64-i686-gcc-libgfortran-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-i686-gcc-libgfortran
mingw32/mingw-w64-i686-gcc-libs 4.9.1-3 (mingw-w64-i686-toolchain mingw-w64-i686) [installed]
    GNU Compiler Collection (libraries) for MinGW-w64
mingw32/mingw-w64-i686-gcc-libs-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-i686-gcc-libs
mingw32/mingw-w64-i686-gcc-objc 4.9.1-3 (mingw-w64-i686-toolchain mingw-w64-i686)
    GNU Compiler Collection (ObjC,Obj-C++) for MinGW-w64
mingw32/mingw-w64-i686-gcc-objc-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-i686-gcc-objc
mingw64/mingw-w64-x86_64-gcc 4.9.1-3 (mingw-w64-x86_64-toolchain mingw-w64-x86_64) [installed]
    GNU Compiler Collection (C,C++,OpenMP) for MinGW-w64
mingw64/mingw-w64-x86_64-gcc-ada 4.9.1-3 (mingw-w64-x86_64-toolchain mingw-w64-x86_64)
    GNU Compiler Collection (Ada) for MinGW-w64
mingw64/mingw-w64-x86_64-gcc-ada-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-x86_64-gcc-ada
mingw64/mingw-w64-x86_64-gcc-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-x86_64-gcc
mingw64/mingw-w64-x86_64-gcc-fortran 4.9.1-3 (mingw-w64-x86_64-toolchain mingw-w64-x86_64)
    GNU Compiler Collection (Fortran) for MinGW-w64
mingw64/mingw-w64-x86_64-gcc-fortran-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-x86_64-gcc-fortran
mingw64/mingw-w64-x86_64-gcc-libgfortran 4.9.1-3 (mingw-w64-x86_64-toolchain mingw-w64-x86_64)
    GNU Compiler Collection (libgfortran) for MinGW-w64
mingw64/mingw-w64-x86_64-gcc-libgfortran-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-x86_64-gcc-libgfortran
mingw64/mingw-w64-x86_64-gcc-libs 4.9.1-3 (mingw-w64-x86_64-toolchain mingw-w64-x86_64) [installed]
    GNU Compiler Collection (libraries) for MinGW-w64
mingw64/mingw-w64-x86_64-gcc-libs-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-x86_64-gcc-libs
mingw64/mingw-w64-x86_64-gcc-objc 4.9.1-3 (mingw-w64-x86_64-toolchain mingw-w64-x86_64)
    GNU Compiler Collection (ObjC,Obj-C++) for MinGW-w64
mingw64/mingw-w64-x86_64-gcc-objc-debug 4.9.1-3
    Detached debugging symbols for mingw-w64-x86_64-gcc-objc
msys/gcc 4.8.2-2 (msys2-devel)
    The GNU Compiler Collection - C and C++ frontends
msys/gcc-fortran 4.8.2-2 (msys2-devel)
    Fortran front-end for GCC
msys/gcc-libs 4.8.2-2 (base) [installed]
    Runtime libraries shipped by GCC
msys/mingw-w64-cross-gcc 4.8.2-1 (mingw-w64-cross-toolchain mingw-w64-cross)
    GCC for the MinGW-w64
```

Is there something I have to take into account when trying to compile ssh? I really want to compile sshd so I can login from my linux machine. That would make development a lot easier.

**Comment by bagong** [comment]
I tried to replicate this and got stuck at authentication. Didn't know which password to use. The Windows user password didn't work, nothing else I tried worked.
Then I tried to replicate the pulic ssh key approach. The client is a macbook trying to connect to msys2.
I copied my public rsa key on the macbook to the file authorized_keys in .ssh in the msys2 home folder (I had to create the file).
Then I tried to connect, using -v, I'll post the output:

```
clientHost:~ username$ ssh -v 10.0.1.28
OpenSSH_6.2p2, OSSLShim 0.9.8r 8 Dec 2011
debug1: Reading configuration data /etc/ssh_config
debug1: /etc/ssh_config line 20: Applying options for *
debug1: /etc/ssh_config line 102: Applying options for *
debug1: Connecting to 10.0.1.28 [10.0.1.28] port 22.
debug1: Connection established.
debug1: identity file /Users/username/.ssh/id_rsa type 1
debug1: identity file /Users/username/.ssh/id_rsa-cert type -1
debug1: identity file /Users/username/.ssh/id_dsa type -1
debug1: identity file /Users/username/.ssh/id_dsa-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_6.2
debug1: Remote protocol version 2.0, remote software version OpenSSH_6.7
debug1: match: OpenSSH_6.7 pat OpenSSH*
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: server->client aes128-ctr hmac-sha1-etm@openssh.com none
debug1: kex: client->server aes128-ctr hmac-sha1-etm@openssh.com none
debug1: SSH2_MSG_KEX_DH_GEX_REQUEST(1024<2048<8192) sent
debug1: expecting SSH2_MSG_KEX_DH_GEX_GROUP
debug1: SSH2_MSG_KEX_DH_GEX_INIT sent
debug1: expecting SSH2_MSG_KEX_DH_GEX_REPLY
debug1: Server host key: RSA 33:0e:34:7a:ec:38:60:3e:5d:77:46:c8:63:44:ce:a1
debug1: Host '10.0.1.28' is known and matches the RSA host key.
debug1: Found key in /Users/username/.ssh/known_hosts:10
debug1: ssh_rsa_verify: signature correct
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: Roaming not allowed by server
debug1: SSH2_MSG_SERVICE_REQUEST sent
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /Users/username/.ssh/id_rsa
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Offering RSA public key: /Users/username/.ssh/id_rsa
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Trying private key: /Users/username/.ssh/id_dsa
debug1: read PEM private key done: type RSA
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: keyboard-interactive
debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: password
username@10.0.1.28's password:
```

I broke off at the password prompt.

What can I do? Are any configurations settings required in /etc/ssh/sshd_config. A weir thing is that my macbook keys were generated as id_rsa and id_rsa.pub while the negotiation with msys seems to use id_rsa id_dsa. But renaming the files didn't change anything...

Any ideas?

**Comment by elieux** [comment]
Just to be sure, is your Mac username the same as your username on Windows? You can list usernames using the `net user` command. I just checked Cygwin's sshd and I was able to connect successfully.

**Comment by bagong** [comment]
It happens to be, but I am pretty sure now that that's not relevant. I've since succeeded by following these instructions:

https://ghc.haskell.org/trac/ghc/wiki/Building/Windows/SSHD

After this was done though, the login problem still persisted. So I can't really tell which or if any of the steps taken above made the difference. What finally made it work for me was creating a user-password for my Windows account *in msys2*. I did it exactly the way given in above link:

```
pwd_entry="$(/usr/bin/mkpasswd -l -u "${username}" | /usr/bin/sed -n -e '/^'${username}'/s?\(^[^:]*:[^:]*:[^:]*:[^:]*:[^:]*:\).*?\1'/var'/empty:/bin/false?p')"
echo "${pwd_entry}" >> "/etc/passwd" 
```

But I replaced ${username} with my username, /var/empty with my home folder, and /bin/false with /bin/bash.

After that I was automatically logged in when knocking on the door with ssh from my mac. I assume that (automatic login, no password check) occured because in earlier attempts I had already added my client host public key to ~/.ssh/authorized_keys and ssh_known_hosts, but no time to double check atm. Note, that I had the keys in the two ssh files before I had added my password-entry in /etc/passwd, and my login was rejected. I realized it had something to do with the existence of these password entries after I found the msys-host would consider a login if I tried to log in as a user that had an password entry (cyg_server and sshd) whereas any other proposed login names were rejected before the password prompt. Likely that was because no home-folder and shell are defined for those two accounts.

If somebody has the time to crosscheck these findings, and if she confirms that the editrights steps are required, this tool should urgently be imported into msys. An sshd that doesn't autostart is only half the fun.

Best
.r
